### ISSUE TITLE
Mono repo support

### ISSUE THREAD

**Issue by johnkors** [Azure/static-web-apps#11]
My scenario: I’m deploying two apps: Sanity studio, a cms admin UI, as well as the frontend (next.js) accessing the headless CMS apis of Sanity. 

I cannot create two sites in Azure Linked to the same repo, due to some restrictions (the error message does not indicate how I can fix it, or if possible at all).

John Papa mentioned there is a workaround using different branches, but it would be better if both apps were following the master branch, so I don’t have to adjust my git flow for the sake of this working.

Thanks!

**Comment by johnpapa** [comment]
Thanks for adding this @johnkors !

I also have a monorepo and I'd love to deploy multiple projects from it, from different folders. It would be nice to have a CI/CD process that would allow for building one or the other, at a time.

Some questions for you:

Would you want to deploy it all at once?

Would you want to deploy one app by itself, and not others in the monorepo?

What would trigger builds?

**Comment by johnkors** [comment]
As far as I understood, [GitHub actions already supports trigger on file paths](https://github.blog/changelog/2019-09-30-github-actions-event-filtering-updates/), so it's just about having several github action ymls for each app.

app1-deploy.yml
```
on:
  push
    paths:
      - "app1/**"
```

app2-deploy.yml
```
on:
  push
    paths:
      - "app2/**"
```

**Comment by johnpapa** [comment]
Here is a sample monorepo project we could use for testing of many apps https://github.com/johnpapa/hello-worlds

**Comment by 0gust1** [comment]
I second this request : we have a monorepo too, and trying to create a second SWA tied to the repo doesn't work. Same ideas and feedbacks as @johnkors

**Comment by k-miyake** [comment]
In front-end development, we are increasingly using monorepo. For example, if you want to share components between the end-user app and the admin app. So I want this Issue to be a higher priority.

**Comment by mtsmachado8** [comment]
Same problem. Iam trying to deploy two web static applications sharing components in a monorepo approach. Azure wont let me do so, didnt find any workaround so i'll have to break my monorepo or switch to another platform... :(

**Comment by nataliaw** [comment]
Same problem :(

**Comment by miwebst** [comment]
We have monorepo support now for all newly created Static Web Apps! You may run into an issue if you have a previously created (before this week) Static Web App attached to the repo/branch you want to use for your monorepo. If you run into that issue, just delete the old Static Web App and the newly created one(s) should then be unblocked.

**Comment by johnpapa** [comment]
@miwebst - Thank you! Can you share how this support works and what settings in the workflow file would change for this?

Will the creation experience change in some way?

**Comment by miwebst** [comment]
So you can customize your workflow to your needs as described above. All that has changed is that you can add multiple Static Web Apps to the same repository + branch. The experience from there is that when you push to that branch (or create a pull request) both sites will get updated/have a staging environment, but you can edit that by specifying which paths to consider in each workflow file.

No change to the creation experience!

**Comment by k-miyake** [comment]
@miwebst Excellent! Thanks for the quick response. Now I can move forward with my product.
